### PR TITLE
Fixed #15664 - Cancel upload file request

### LINKED ISSUE #15664: FileUpload: Never cancels the upload request when I click on the cancel button.
### Describe the bug

If I upload a (large) file and cancel it before it has been uploaded, the HTTP upload request is never canceled. 

### Environment

Doesnt matter.

### Reproducer

https://stackblitz.com/edit/zjfynw

### Angular version

17.3.7

### PrimeNG version

17.17.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

 v18.20.3

### Browser(s)

_No response_

### Steps to reproduce the behavior

https://stackblitz.com/edit/zjfynw

Add file for upload 
01-add-file-for-upload

Click on upload button
02-click-upload-button

Click on remove file button
03-click-cancel-upload

It continues uploading
04-continue-uploading

And it finishes uploading.
05-finishes-uploading

### Expected behavior

When I click on the remove file button, it unsubscribes from the HTTP request.
06-http-request-canceled

### PR DESCRIPTION
**Pull request by kubedan**
Fixes #15664

### CODE REVIEW

File: src/app/components/fileupload/fileupload.ts [hunk 1]
```diff
@@ -826,12 +833,22 @@ export class FileUpload implements AfterViewInit, AfterContentInit, OnInit, OnDe
...
+     removeUploadedFile(index) {
         let removedFile = this.uploadedFiles.splice(index, 1)[0];
         this.uploadedFiles = [...this.uploadedFiles];
         this.onRemoveUploadedFile.emit({ file: removedFile, files: this.uploadedFiles });
     }
 
+    /**
+     * Cancel upload file request.
+     * */
+    cancelUploadRequst() {
```
Comment (SoyDiego) [inline src/app/components/fileupload/fileupload.ts:845]: typo error, should be `cancelUploadRequest`

Comment (kubedan) [reply src/app/components/fileupload/fileupload.ts]: Thx, I fixed it.

**Comment by cetincakiroglu** [comment]
Hi,

Could you please link an issue to this PR

**Comment by kubedan** [comment]
> Hi,
> 
> Could you please link an issue to this PR

Done

https://github.com/primefaces/primeng/pull/15668#issue-2312713149

**Pull request merged**